### ISSUE TITLE
Fix false positives for namespaced variables in property-no-unknown

### ISSUE THREAD

**Issue by johncrim** [stylelint/stylelint#4544]
It appears that the `property-no-unknown` rule is incorrectly flagging SCSS `@use` namespaces, when setting a variable within a namespace. This occurs only within `@mixin`s and `@function`s, it doesn't occur when reading or writing module properties in the global scope. It also only occurs while setting a variable - it doesn't occur when reading the variable.

SCSS `@use` and module namespaces are relatively new - [they're described here](https://sass-lang.com/documentation/at-rules/use#loading-members).

```scss
@use 'accoutrement/sass' as a;
@use 'sass:list';

// No stylelint error
a.$sizes: ();

@function clear-sizes() {
  // Lint error: Unexpected unknown property "a.$sizes"   property-no-unknown
  a.$sizes: ();
  @return a.$sizes; // No lint error
}
$s: clear-sizes();

@mixin dump-sizes() {
  // No stylelint error
  $sizes: a.$sizes;
  @debug '# of sizes registered: #{ list.length($sizes) }';
}
@include dump-sizes();
```

The linked file provides a simple repro case with dependencies.
[stylelint-module-property.zip](https://github.com/kristerkari/stylelint-scss/files/4037883/stylelint-module-property.zip)

> Which rule, if any, is the bug related to?

property-no-unknown

> What CSS is needed to reproduce the bug?

```scss
@use 'accoutrement/sass' as a;

@function clear-sizes() {
  // Lint error: Unexpected unknown property "a.$sizes"   property-no-unknown
  a.$sizes: ();
  @return a.$sizes; // No error
}
```

> What stylelint configuration is needed to reproduce the bug?

stylelint.config.js
```js
module.exports = {
  rules: {
    'property-no-unknown': true,
  }
};
```

> Which version of stylelint are you using?

12.0.1

> How are you running stylelint: CLI, PostCSS plugin, Node.js API?

CLI: `stylelint **/*.scss`

> Does the bug relate to non-standard syntax (e.g. SCSS, Less etc.)?

Yes, it's related to SCSS module namespaces.

> What did you expect to happen?

No lint error.

> What actually happened (e.g. what warnings or errors did you get)?

```shell
$ stylelint **/*.scss

src/stylelint-use-namespace.scss
 9:3  ×  Unexpected unknown property "a.$sizes"   property-no-unknown

error Command failed with exit code 2.
```

I originally posted this bug [in the stylelint-scss project](https://github.com/kristerkari/stylelint-scss/issues/426), but I was directed to post it here.

**Comment by jeddy3** [comment]
@johncrim Thanks for the comprehensive report.

I [can replicate](https://stylelint.io/demo#N4Igxg9gJgpiBcIACAzArgOzAFwJYQwAIAPACgEpDhCBbAOgBIA3AQwCd5CKBuQgXwA6GEABoQAZwCeGbC2IJw48aPAEUuAOYLgQwoQEg2aADYxxBzjqJ79IAA5sIdmG2ySAtBgjvMAay8A7hgWhNhGMLr8QnwgfEA) the issue with:

```scss
@function x() { m.$var: (); }
```

PostCSS [parses this construct as a declaration with a property of `m.$var`](https://astexplorer.net/#/gist/db35abf93955a99a2075f2a3cce888aa/2bf194c2e391d2b1b8654a10c0f257ad3d543a32), understandably. 

I believe we can fix the issue by adding a new conditional in [`isStandardSyntaxDeclaration`](https://github.com/stylelint/stylelint/blob/master/lib/utils/isStandardSyntaxDeclaration.js) to ignore this construct.

I suspect we can return `false` if the `prop` of the `decl` contains `.$`. Can anyone foresee any false negative problems with this?

**Comment by jeddy3** [comment]
I am labelling as ready to implement. Feel free to contribute a PR to fix this.
